packages/core/src/Repo.ts

```diff
@@ -270,20 +243,48 @@ export class Repo<T = any> {
     return history
   }
 
-  /**
-   * Used when receiving the entire current state of a repo from a peer.
-   */
+  /** Used when receiving the entire current state of a repo from a peer. */
   async loadHistory(history: RepoHistory) {
     for (const documentId in history) {
       const changes = history[documentId]
       await this.applyChanges(documentId, changes)
     }
   }
 
+  /**
+   * Accessor for a document's clock
+   * @param documentId
+   * @returns Our clock, or if none exists, an empty clock
+   */
+  public getClock(documentId: string) {
+    return this.clock[documentId] || EMPTY_CLOCK
+  }
+
+  /** Returns true if we have a clock in memory for this document */
+  public hasClock(documentId: string) {
+    return this.clock.hasOwnProperty(documentId)
+  }
+
+  /** Returns our entire ClockMap as-is */
+  public getClocks() {
+    return this.clock
+  }
+
+  /**
+   * Updates the vector clock by merging in the new vector clock `clock`, setting each node's
+   * sequence number to the maximum for that node
+   * @param documentId
+   * @param newClock
+   */
+  public updateClock(documentId: string, newClock: Clock) {
+    const oldClock = this.clock[documentId]
+    this.clock[documentId] = mergeClocks(oldClock, newClock)
+  }
+
```

Comment:
The public API for managing "our" clocks: get one, check for existence, get all, and update. 

Note that `updateClock` does a merge, not a simple replacement: We preserve the highest value for each actor.